src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +62,642 @@ const ActorInit En_Rz_InitVars = {
+            }
+            EnRz_ChangeAnim(play, this, EN_RZ_ANIM_3, ANIMMODE_LOOP, 0.0f);
+            this->actionFunc = func_80BFC674;
+            this->actor.shape.yOffset = -1500.0f;
+            break;
+
+        case EN_RZ_TYPE_2:
+            this->actor.flags |= ACTOR_FLAG_10;
+            if (gSaveContext.save.weekEventReg[77] & 4) {
+                EnRz_ChangeAnim(play, this, EN_RZ_ANIM_8, ANIMMODE_LOOP, 0.0f);
+            } else {
+                EnRz_ChangeAnim(play, this, EN_RZ_ANIM_6, ANIMMODE_LOOP, 0.0f);
+            }
+            this->actionFunc = func_80BFC3F8;
+            this->sister = EnRz_FindSister(this, play);
+            this->actor.uncullZoneForward = 300.0f;
+            break;
+
+        default:
+            this->stateFlags |= EN_RZ_STATE_2;
+            if (gSaveContext.save.weekEventReg[77] & 4) {
+                Actor_MarkForDeath(&this->actor);
+                return;
+            }
+            EnRz_ChangeAnim(play, this, EN_RZ_ANIM_2, ANIMMODE_LOOP, 0.0f);
+            this->actionFunc = func_80BFC8F8;
+            if (!EN_RZ_GET_SISTER(thisx)) {
+                this->actor.textId = 0x291C;
+            } else {
+                this->actor.textId = 0x2920;
+            }
+            break;
+    }
+    if (!EN_RZ_GET_SISTER(thisx)) {
+        this->csAction = 0x226;
+    } else {
+        this->csAction = 0x227;
+    }
+
+    this->actionIndex = 0;
+}
+
+/**
+ * Custom shadow draw function of type ActorShadowFunc.
+ */
+void EnRz_ActorShadowFunc(Actor* thisx, Lights* mapper, PlayState* play) {
+    Vec3f oldPos;
+    EnRz* this = THIS;
+
+    if (this->animIndex == EN_RZ_ANIM_8) {
+        f32 tempScale = (((27.0f - this->shadowPos.y) + this->actor.world.pos.y) * ((1 / 2.25f) * 0.001f)) + 0.01f;
+
+        this->actor.scale.x = tempScale;
+    }
+
+    Math_Vec3f_Copy(&oldPos, &this->actor.world.pos);
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->shadowPos);
+    func_800B4AEC(play, &this->actor, 50.0f);
+
+    if (oldPos.y < this->actor.floorHeight) {
+        this->actor.world.pos.y = this->actor.floorHeight;
+    } else {
+        this->actor.world.pos.y = oldPos.y;
+    }
+
+    ActorShadow_DrawCircle(&this->actor, mapper, play);
+    Math_Vec3f_Copy(&this->actor.world.pos, &oldPos);
+    this->actor.scale.x = 0.01f;
+}
+
+void EnRz_ChangeAnim(PlayState* play, EnRz* this, s16 animIndex, u8 animMode, f32 transitionRate) {
+    static AnimationHeader* D_80BFCD20[] = {
+        &object_rz_Anim_00457C, &object_rz_Anim_003A20, &object_rz_Anim_005E50, &object_rz_Anim_003098,
+        &object_rz_Anim_00059C, &object_rz_Anim_000DE8, &object_rz_Anim_0028D4,
+    };
+
+    static AnimationHeader* D_80BFCD3C[] = {
+        &object_rz_Anim_003A20, &object_rz_Anim_003A20, &object_rz_Anim_005390, &object_rz_Anim_003098,
+        &object_rz_Anim_00059C, &object_rz_Anim_000DE8, &object_rz_Anim_0028D4,
+    };
+
+    static LinkAnimationHeader* sLinkAnimations[] = { &gPlayerAnim_link_normal_wait_free,
```

Comment:
```suggestion
    static AnimationHeader* D_80BFCD3C[] = {
        &object_rz_Anim_003A20, &object_rz_Anim_003A20, &object_rz_Anim_005390, &object_rz_Anim_003098,
        &object_rz_Anim_00059C, &object_rz_Anim_000DE8, &object_rz_Anim_0028D4,
    };
    static LinkAnimationHeader* sLinkAnimations[] = { &gPlayerAnim_link_normal_wait_free,
```
This data takes up enough room without extra newlines.